internal/configfsm/fsm.go

```diff
@@ -0,0 +1,309 @@
+package configfsm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/qmuntal/stateless"
+)
+
+type Trigger string
+
+const (
+	// Transitions to Missing state
+	StartToMissing           Trigger = "Start->Missing"
+	ClusterNotReadyToMissing Trigger = "ClusterNotReady->Missing"
+	InProgressToMissing      Trigger = "InProgress->Missing"
+	CompletedToMissing       Trigger = "Completed->Missing"
+	TimedOutToMissing        Trigger = "TimedOut->Missing"
+	OutOfDateToMissing       Trigger = "OutOfDate->Missing"
+
+	// Transitions to ClusterNotReady state
+	MissingToClusterNotReady    Trigger = "Missing->ClusterNotReady"
+	InProgressToClusterNotReady Trigger = "InProgress->ClusterNotReady"
+	CompletedToClusterNotReady  Trigger = "Completed->ClusterNotReady"
+	OutOfDateToClusterNotReady  Trigger = "OutOfDate->ClusterNotReady"
+
+	// Transitions to InProgress state
+	ClusterNotReadyToInProgress Trigger = "ClusterNotReady->InProgress"
+	OutOfDateToInProgress       Trigger = "OutOfDate->InProgress"
+	CompletedToInProgress       Trigger = "Completed->InProgress"
+
+	// Transitions to TimedOut state
+	ClusterNotReadyToTimedOut Trigger = "ClusterNotReady->TimedOut"
+	InProgressToTimedOut      Trigger = "InProgress->TimedOut"
+
+	// Transitions to Completed state
+	InProgressToCompleted Trigger = "InProgress->Completed"
+
+	// Transitions to OutOfDate state
+	InProgressToOutOfDate      Trigger = "InProgress->OutOfDate"
+	TimedOutToOutOfDate        Trigger = "TimedOut->OutOfDate"
+	ClusterNotReadyToOutOfDate Trigger = "ClusterNotReady->OutOfDate"
+	CompletedToOutOfDate       Trigger = "Completed->OutOfDate"
```

Comment:
`Missing` -> `OutOfData` is also a possibility if we have no policies, but then we add policies in `inform`.